cmd/p2p/crawl/crawl.go

```diff
@@ -56,57 +58,75 @@ var CrawlCmd = &cobra.Command{
 		return nil
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
-		nodes, err := p2p.ReadNodeSet(inputCrawlParams.NodesFile)
-		if err != nil {
-			log.Warn().Err(err).Msgf("Creating nodes file %v because it does not exist", inputCrawlParams.NodesFile)
-		}
-
-		var cfg discover.Config
-		cfg.PrivateKey, _ = crypto.GenerateKey()
-		cfg.Bootnodes, err = p2p.ParseBootnodes(inputCrawlParams.Bootnodes)
-		if err != nil {
-			return fmt.Errorf("unable to parse bootnodes: %w", err)
-		}
-
-		db, err := enode.OpenDB(inputCrawlParams.Database)
-		if err != nil {
-			return err
-		}
-
-		ln := enode.NewLocalNode(db, cfg.PrivateKey)
-		socket, err := p2p.Listen(ln)
-		if err != nil {
-			return err
+		switch {
+		// Check that one and only one of --bootnodes or --discovery-dns flag is being used
+		case inputCrawlParams.Bootnodes != "" && inputCrawlParams.DiscoveryDNS != "":
+			log.Fatal().Msg("Use either --bootnodes flag or --discovery-dns flag")
+			return nil
+		// Check that at least one of --bootnodes or --discovery-dns flag is being used
+		case inputCrawlParams.Bootnodes == "" && inputCrawlParams.DiscoveryDNS == "":
+			log.Fatal().Msg("Use only one of --bootnodes flag or --discovery-dns flag")
+			return nil
+		// Discovery DNS mode
```

Comment:
for this instead you could use `MarkFlagsMutuallyExclusive` see example: https://github.com/0xPolygon/polygon-cli/blob/main/cmd/p2p/sensor/sensor.go#L497